src/util/Exception.cxx

```diff
@@ -1,99 +1,214 @@
-      logger << "\tat " << symbol.Name << "\n";
-  }
-  exit(1);
+    void *array[100];
+    int size = backtrace(array, 100);
+
+    char **stack = backtrace_symbols(array, size);
+
+    for (int i = 0; i < size; i++) {
+        LOG(LOG_ERROR) << stack[i];
+    }
 }
 
-#elif (defined(__unix__) || defined(__APPLE__)) && !defined(__ANDROID__) && !defined(__EMSCRIPTEN__)
+static void LwCrashHandler(int sig)
+{
+    SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Oops, crashed with signal %d :(", sig);
+    LwBacktracePrint();
+    exit (1);
+}
 
-#include <unistd.h>
-#include <execinfo.h>
-#include <signal.h>
+void SystemSetupCrashHandler(void)
+{
+    signal(SIGSEGV, LwCrashHandler);
+}
 
-void SIG_handler(int signal)
+#elif defined(_WIN32) && defined(_M_X64)
+
+#include <windows.h>
+#include <imagehlp.h>
+#include <stdio.h>
+
+static const char *LwPrintExceptionName(DWORD exception_code)
 {
-  switch (signal)
-  {
-  case SIGINT:
-    [[fallthrough]];
-  case SIGTERM:
+  switch (exception_code)
   {
-    LOG(LOG_INFO) << "Abort signal received";
-    exit(0);
+  case EXCEPTION_ACCESS_VIOLATION:
+    return "Access violation";
+  case EXCEPTION_DATATYPE_MISALIGNMENT:
+    return "Datatype misalignment";
+  case EXCEPTION_BREAKPOINT:
+    return "Breakpoint";
+  case EXCEPTION_SINGLE_STEP:
+    return "Single step";
+  case EXCEPTION_ARRAY_BOUNDS_EXCEEDED:
+    return "Array bounds exceeded";
+  case EXCEPTION_FLT_DENORMAL_OPERAND:
+    return "Float denormal operand";
+  case EXCEPTION_FLT_DIVIDE_BY_ZERO:
+    return "Float divide by zero";
+  case EXCEPTION_FLT_INEXACT_RESULT:
+    return "Float inexact result";
+  case EXCEPTION_FLT_INVALID_OPERATION:
+    return "Float invalid operation";
+  case EXCEPTION_FLT_OVERFLOW:
+    return "Float overflow";
+  case EXCEPTION_FLT_STACK_CHECK:
+    return "Float stack check";
+  case EXCEPTION_FLT_UNDERFLOW:
+    return "Float underflow";
+  case EXCEPTION_INT_DIVIDE_BY_ZERO:
+    return "Integer divide by zero";
+  case EXCEPTION_INT_OVERFLOW:
+    return "Integer overflow";
+  case EXCEPTION_PRIV_INSTRUCTION:
+    return "Privileged instruction";
+  case EXCEPTION_IN_PAGE_ERROR:
+    return "In page error";
+  case EXCEPTION_ILLEGAL_INSTRUCTION:
+    return "Illegal instruction";
+  case EXCEPTION_NONCONTINUABLE_EXCEPTION:
+    return "Noncontinuable exception";
+  case EXCEPTION_STACK_OVERFLOW:
+    return "Stack overflow";
+  case EXCEPTION_INVALID_DISPOSITION:
+    return "Invalid disposition";
+  case EXCEPTION_GUARD_PAGE:
+    return "Guard page";
+  case EXCEPTION_INVALID_HANDLE:
+    return "Invalid handle";
   }
-  case SIGSEGV:
-  {
-    LOG(LOG_ERROR) << "Segmentation Fault";
-    break;
-  }
-  default:
+  return "Unknown exception";
+}
+
+static void LwSystemError(const char *title, const char *text)
+{
+  MessageBox(NULL, text, title, MB_OK | MB_ICONERROR);
+}
+
+static void LwPrintStacktrace(LPEXCEPTION_POINTERS e)
+{
+  char crash_info[256];
+  int filelineinfo_ok = 0;
```

Comment:
Consider defining variable in the place of the first usage, I find code easier to read in that case

Reply:
pls save this code in this state, it common example from microsoft docs, if something will be changed I can see it, but if code different from example it need much time to analize

Reply:
OK, could you please add a comment with reference to the example page?